### DISCUSSION TITLE
plot with charge power vs SOC

### DISCUSSION THREAD

**Discussion by fzahle** [teslamate-org/teslamate#849] [Ideas]
**What would you like to be added**:
I would like to be able to plot a scatter plot like in this link including all super charging sessions: https://forum.abetterrouteplanner.com/blogs/entry/30-tesla-supercharging-summer-2019-update/.

![BT37_3long png c320c6b80392c1e7927cde3b46570851](https://user-images.githubusercontent.com/1467214/88341234-8669c800-cd3d-11ea-8b52-9d6530330c97.png)

**Why is this needed**:
It is currently not possible to graphically compare charging sessions. It could be neat to have a plot like this together with the main Charges table, but also in the charge details for the chosen session.

**Additional context**:
I tried to play with Grafana, but quickly realised that Grafana really only does time series. There is a Plotly plugin that would allow scatter plots with a different x-axis, but I don't yet have TeslaMate set up to develop locally. So if someone else likes the idea and is faster at setting something like this up it would be great!

**Comment by Dulanic** [comment]
It wouldnt look nearly like that, because we have so little data for each of our own.

Example: I laid this out quickly with tableau just to quickly show what it would look like. To me that has no value added?

https://dulanic.com/images/AssuredLiquidAntarcticgiantpetrel.png

![image](https://user-images.githubusercontent.com/2948206/88571524-68cd8480-d003-11ea-8a38-349f53e469ca.png)

**Comment by fzahle** [comment]
the plot you show is not quite what I'm looking for though. It should simply be a scatter or lines plot with all data from the charges with SOC on the x-axis and power on the y-axis.

**Comment by Dulanic** [comment]
That's what that is. A scatter plot. It looks like a line because all my charges are at home, same kW.  That's my point, when it's just one person it does very little.

**Comment by GtHbbrr** [comment]
can you filter on the DC charges only?

**Comment by Dulanic** [comment]
I found another method, I make no promises I can duplicate in grafana, though I don't see why not. The problem still comes in that you're working with limited data when it's just you.  ABRP is using the data from hundreds or even thousands of vehicles data.

https://public.tableau.com/profile/dulanic#!/vizhome/TeslaChargeScatterPlot/ChargeScatterPlot

**Comment by fzahle** [comment]
That’s it! You’re right that each of us has limited data, but I still think it could be quite informative, in particular if it could be possible to interact with the data in the graph when hovering over it. 

One way could be to make line plots for each charging session with very dim lines as well as scatter circles as you’ve done now and then if possible highlight a given charging curve when hovering over its data with some info on total energy added, total time etc, essentially the info in the table. This is all doable in plot.ly but don’t know about grafana. I will try to see if I can connect to the database from my laptop and play with a Jupiter notebook. So far, I’ve only been able to connect with the postgresql container from the host raspberrypi which is less convenient. 

Edit: didn’t realize that you already made the plot interactive with info on total energy added etc., great!

**Comment by Dulanic** [comment]
So I got it built out /w grafana and I am not a fan. I can't get the plot nearly as interactive nor can I figure out how to do a trendline /w grafana and the Plotly plugin.



![image](https://user-images.githubusercontent.com/2948206/88707156-7c4a1f80-d0d7-11ea-9963-48d12bd7318b.png)

**Comment by fzahle** [comment]
I now extracted the data I have from charging sessions and it looks like this plotted with a simple python matplotlib scatter plot. I only have 11 supercharging sessions in there, and even then I think this simple plot even without any interactive features would be quite useful to track overall trends in supercharging.

![charging_sessions](https://user-images.githubusercontent.com/1467214/88714701-75460000-d11d-11ea-8d5b-02e32240ebec.png)

**Comment by Dulanic** [comment]
Feel free to play with my grafana, if has the plug in added.

docker pull dulanic/grafana:latest

**Comment by GtHbbrr** [comment]
I tried updating my teslamate instance with your grafana update but cannot get it to work. I tried your suggestion for docker dulanic/grafana:latest but I dont see any new dashboards. What am I missing?

**Comment by BabyYeggie** [comment]
> It wouldnt look nearly like that, because we have so little data for each of our own.
> 
> Example: I laid this out quickly with tableau just to quickly show what it would look like. To me that has no value added?
> 
> https://dulanic.com/images/AssuredLiquidAntarcticgiantpetrel.png

Which postgres gui are you using? That looks better than PGAdmin4.

**Comment by rawdigits** [comment]
I used the Plot.ly grafana plugin to do basically what you wanted here. This is 2 years of supercharging on my model X.

It is a bit funky because SOC is tracked by whole numbers here. Also, I just used opacity in the color code to allow the trends to appear as somewhat darker when there are more data points.

It is inelegant, but works. Happy to share the queries/etc if anyone else finds this useful. It taught me that my X has actually hit a higher peak supercharging rate than my 3 ever has. 😄 

<img width="2836" alt="Screen Shot 2020-10-30 at 4 18 01 AM" src="https://user-images.githubusercontent.com/1096409/97699443-626a0100-1a67-11eb-9fac-04e33914a49f.png">

**Comment by GtHbbrr** [comment]
Nice! I have no experience with Plot.ly but my experience with Excel graphs to allow trends to appear somewhat darker I use open dots instead of your dashes for data point to show trend without individual data becoming invisible.

**Comment by rawdigits** [comment]
I just played around with different shapes, but some of them just create a blurry mess that gives me a headache. I'm definitely out of my wheelhouse at this point, unfortunately.

One pretty cool thing I was able to do was compare the charge curve from jan-april against april-now, and you can *definitely* see they have raised it in interim updates.

**Comment by Dulanic** [comment]
> > It wouldnt look nearly like that, because we have so little data for each of our own.
> > Example: I laid this out quickly with tableau just to quickly show what it would look like. To me that has no value added?
> > https://dulanic.com/images/AssuredLiquidAntarcticgiantpetrel.png
> 
> Which postgres gui are you using? That looks better than PGAdmin4.

Missed this, I am using PGAdmin. I can't see the pic I used before because I purge all screenshots on a rolling 30 days. Next time Ill just paste direct to github. I do use dark mode, but thats about it.
```
  pgadmin:
    image: dpage/pgadmin4:latest
```

As for the rest of this, yeah I kind of gave up since I couldn't get it to look nicer.
